ai-service/src/ai_service/project_ingestor.py

```
@@ -0,0 +1,60 @@
+    ".md",
+    ".yml",
+}
+
+
+def clone_github_repo(repo_url: str) -> str:
+    """
+    Clones a GitHub repo to a temporary directory.
+    Returns the path to the cloned directory.
+    """
+    clone_to = tempfile.mkdtemp()
+    subprocess.run(["git", "clone", repo_url, clone_to], check=True)
```

Ruff (0.12.2)

`subprocess` call: check for execution of untrusted input

(S603)